extensions/wikia/Email/EmailController.class.php

```diff
@@ -0,0 +1,279 @@
+			'msg' => $e->getMessage(),
+		] );
+		return true;
+	}
+
+	protected function getTargetLang() {
+		return $this->targetUser->getOption( 'language' );
+	}
+
+	/**
+	 * Returns the address for the recipient of this email
+	 *
+	 * @return \MailAddress
+	 */
+	protected function getToAddress() {
+		$to = new \MailAddress( $this->targetUser->getEmail(), $this->targetUser->getName() );
+		return $to;
+	}
+
+	/**
+	 * Returns the address for the sender of this email
+	 *
+	 * @return \MailAddress
+	 */
+	protected function getFromAddress() {
+		$sender = new \MailAddress( $this->wg->PasswordSender, $this->wg->PasswordSenderName );
+		return $sender;
+	}
+
+	/**
+	 * Get the 'reply-to' address for the send of this email, if different than the 'from' address.
+	 *
+	 * @return \MailAddress|null
+	 */
+	protected function getReplyToAddress() {
+		$replyAddr = null;
+		if ( !empty( $this->wg->NoReplyAddress ) ) {
+			$name = wfMessage('emailext-no-reply-name' )->escaped();
+			$replyAddr = new \MailAddress( $this->wg->NoReplyAddress, $name );
+		}
+
+		return $replyAddr;
+	}
+
+	/**
+	 * Return the subject used for this email
+	 */
+	abstract protected function getSubject();
+
+	/**
+	 * Renders the 'body' view of the current email controller
+	 *
+	 * @return string
+	 */
+	protected function getBody() {
+		$html = $this->app->renderView(
+			get_class( $this ),
+			'body',
+			$this->request->getParams()
+		);
+		return $html;
+	}
+
+	protected function findUserFromRequest( $paramName, \User $default = null ) {
+		$userName = $this->getRequest()->getVal( $paramName );
+		if ( empty( $userName ) ) {
+			return $default;
+		}
+
+		return $this->getUserFromName( $userName );
+	}
+
+	/**
+	 * Return a user object from a username
+	 *
+	 * @param string $username
+	 *
+	 * @return \User
+	 * @throws \Email\Fatal
+	 */
+	protected function getUserFromName( $username ) {
+		if ( is_null( $username ) ) {
+			throw new Fatal( wfMessage( 'emailext-error-noname' )->escaped() );
+		}
+
+		$user = \User::newFromName( $username );
+		$this->assertValidUser( $user );
+
+		return $user;
+	}
+
+	/**
+	 * A basic check to see if we should send this email or not
+	 *
+	 * @throws \Email\Check
+	 */
+	public function assertCanEmail() {
+		$this->assertUserWantsEmail();
+		$this->assertUserNotBlocked();
+	}
+
+	/**
+	 * @param $user
+	 *
+	 * @throws \Email\Fatal
+	 */
+	public function assertValidUser( $user ) {
+		if ( !$user instanceof \User ) {
+			throw new Fatal( wfMessage('emailext-error-not-user' )->escaped() );
+		}
+
+		if ( $user->getId() == 0 ) {
+			throw new Fatal( wfMessage( 'emailext-error-empty-user' )->escaped() );
+		}
+	}
+
+	/**
+	 * @throws \Email\Fatal
+	 */
+	public function assertUserHasEmail() {
+		$email = $this->targetUser->getEmail();
+		if ( empty( $email ) ) {
+			throw new Fatal( wfMessage( 'emailext-error-noemail' )->escaped() );
+		}
+	}
+
+	/**
+	 * @throws \Email\Fatal
+	 */
+	public function assertHasIP() {
```

Comment:
Ah, looks like it might be for the hook run in `ForgotPasswordController`? Can we move this method definition into `ForgotPasswordController` there then, or do we expect this to be used in other places?